client/data/marketplace/use-marketplace-reviews.ts

```diff
@@ -290,12 +290,31 @@ export const useCreateMarketplaceReviewMutation = () => {
 	} );
 };
 
-export const useUpdateMarketplaceReviewMutation = () => {
+export const useUpdateMarketplaceReviewMutation = ( {
+	productType,
+	slug,
+	page,
+	perPage,
+	author,
+	author_exclude,
+}: MarketplaceReviewsQueryProps ) => {
 	const queryClient = useQueryClient();
+	const queryKey: QueryKey = [
+		queryKeyBase,
+		productType,
+		slug,
+		author,
+		author_exclude,
+		page,
+		perPage,
+	];
 	return useMutation( {
 		mutationFn: updateReview,
 		onSuccess: () => {
-			queryClient.invalidateQueries( { queryKey: queryKeyBase } );
+			queryClient.invalidateQueries( { queryKey } );
+		},
+		onError: ( error: Error ) => {
+			alert( error.message );
 		},
```

Comment:
We may want to use the error callback to do this and the client may decide how to display the error.

Reply:
I removed the `onError` call, and added it here: https://github.com/Automattic/wp-calypso/pull/85305/files#diff-72b59db4ba804fcbd1962c7f5515d8e995f414cf742c6dd513d2ef1bbc15dad2R189

This works as expected, but I couldn't resolve a type mismatch:
![CleanShot 2024-01-03 at 16 18 44@2x](https://github.com/Automattic/wp-calypso/assets/11555574/22cdece1-747d-43d9-937a-2e4679b2f5a1)
![CleanShot 2024-01-03 at 16 20 37@2x](https://github.com/Automattic/wp-calypso/assets/11555574/1697900a-506b-49f6-8a2d-b5e1a4471d30)

@WBerredo could you help with how we could add the error type here? Thanks!

Reply:
Passing the correct type for `useMutation` should do the trick.


On `use-marketplace-reviews.ts` line `295`:
```typescript
return useMutation< MarketplaceReviewResponse, ErrorResponse, MarketplaceReviewBody >( {
```

Reply:
Thanks @WBerredo, that really did the trick in https://github.com/Automattic/wp-calypso/pull/85305/commits/558dc914aed5ae680ac05027019841acb906a8a1, CI is green now.